src/main/java/com/example/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,136 @@
+package com.example.onlinebookstore.service.impl;
+
+import com.example.onlinebookstore.dto.OrderItemResponseDto;
+import com.example.onlinebookstore.dto.OrderRequestDto;
+import com.example.onlinebookstore.dto.OrderResponseDto;
+import com.example.onlinebookstore.dto.OrderStatusUpdateDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.exception.UnauthorizedAccessException;
+import com.example.onlinebookstore.mapper.OrderItemMapper;
+import com.example.onlinebookstore.mapper.OrderMapper;
+import com.example.onlinebookstore.model.Order;
+import com.example.onlinebookstore.model.OrderItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.Status;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.OrderItemRepository;
+import com.example.onlinebookstore.repository.OrderRepository;
+import com.example.onlinebookstore.service.OrderService;
+import com.example.onlinebookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart cart = shoppingCartService.findByUserId(user);
+
+        Order order = new Order();
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(orderRequestDto.getShippingAddress());
+
+        Set<OrderItem> orderItems = getOrderItems(cart, order);
+        order.setOrderItems(orderItems);
+
+        BigDecimal total = calculateTotalPrice(orderItems);
+        order.setTotal(total);
+
+        Order savedOrder = orderRepository.save(order);
+        cart.getCartItems().clear();
```

Comment:
I belive you should save cart after clearing cartItems to make flush changes to db